### PR TITLE
add manylinux auditwheel recipe

### PR DESCRIPTION
**Pull request by msarahan**
Not sure if this is the best way to do this, or if we should bundle auditwheel directly in the docker image without building it here.  Open to discussion.

### CODE REVIEW

**Comment by jjhelmus** [comment]
@msarahan I'd be in favor of including this as a conda package and not in the docker image at least for now.  If we find that the tool is used often we can always add it to the docker image later.  

I believe you need to have something in the test section for the linter to not complain.  Checking that the `auditwheel -h` command works should be sufficient.

File: recipes/auditwheel/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  name: auditwheel
+  version: "1.1.0"
+
+source:
+  fn: auditwheel-1.1.0.tar.gz
+  url: https://pypi.python.org/packages/source/a/auditwheel/auditwheel-1.1.0.tar.gz
+  md5: f42ec0ed1ce5e09e5ef5d3ce4f9826e6
+
+build:
+  noarch_python: True
```
Comment (pelson) [inline recipes/auditwheel/meta.yaml]: Out of interest, how come? We don't really want somebody to be able to install auditwheel on Windows (AFAICT) so I'm curious. :smile:

**Comment by pelson** [comment]
:+1: - just a small question about the noarch usage.

File: recipes/auditwheel/meta.yaml (hunk 1 above)
Comment (msarahan) [inline recipes/auditwheel/meta.yaml]: That's very true.  I've gone back and forth on this.  I really don't like conda's general rule of having to build out all packages when a single noarch package suffices (for example, for pure python packages).

If you try to install auditwheel with pip on WIndows, it succeeds.  Running it gives

    Error: This tool only supports Linux

My motivation was really avoiding separate Python builds for Linux.

File: recipes/auditwheel/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  name: auditwheel
+  version: "1.1.0"
+
+source:
+  fn: auditwheel-1.1.0.tar.gz
+  url: https://pypi.python.org/packages/source/a/auditwheel/auditwheel-1.1.0.tar.gz
+  md5: f42ec0ed1ce5e09e5ef5d3ce4f9826e6
+
+build:
+  skip: True  # [not linux]
```
Comment (jjhelmus) [inline recipes/auditwheel/meta.yaml]: auditwheel requires [Python 3.3+](https://github.com/pypa/auditwheel#installation) so adding `skip: True  # [py2k]` would probably be reasonable.   

I checked the dependency on my system, auditwheel will install in Python 2.7, but raises a SyntaxError when `auditwheel` is executed.

Comment (msarahan) [inline recipes/auditwheel/meta.yaml]: Is that a better way than the runtime dependency?

Comment (msarahan) [inline recipes/auditwheel/meta.yaml]: Never mind, just looked at the circle CI traceback.  Your way is definitely better.

**Comment by msarahan** [comment]
Appears to be Python 3.5 only (typing was added in 3.5):

```
+ auditwheel -h
Traceback (most recent call last):
  File "/opt/conda/envs/_test/bin/auditwheel", line 6, in <module>
    from auditwheel.main import main
  File "/opt/conda/envs/_test/lib/python3.4/site-packages/auditwheel/main.py", line 8, in <module>
    from . import main_addtag
  File "/opt/conda/envs/_test/lib/python3.4/site-packages/auditwheel/main_addtag.py", line 2, in <module>
    from .policy import (load_policies, get_policy_name, get_priority_by_name,
  File "/opt/conda/envs/_test/lib/python3.4/site-packages/auditwheel/policy/__init__.py", line 4, in <module>
    from typing import Optional
ImportError: No module named 'typing'
```

**Comment by jjhelmus** [comment]
[typing has been backported](https://pypi.python.org/pypi/typing/), probably need to add this as a dependency for non-python 3.5.

File: recipes/auditwheel/build.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+$PYTHON setup.py install
```
Comment (pelson) [inline recipes/auditwheel/build.sh]: Should this be ```--single-version-externally-managed --record=record.txt```?

Comment (jakirkham) [inline recipes/auditwheel/build.sh]: Should it also go in `meta.yaml` under `build/script`?

**Comment by pelson** [comment]
This looks good to me. I'm merging as is, though I have a question as to whether we are missing a ```--single-version-externally-managed``` flag - if so, we can do that in the feedstock.

File: recipes/auditwheel/meta.yaml (hunk 2 above)
Comment (jakirkham) [inline recipes/auditwheel/meta.yaml]: Could update to `# [not linux and py2k]` later.

**Pull request merged**